python/paddle/fluid/tests/unittests/test_ast_transformer_static_analysis.py

```diff
@@ -81,15 +81,48 @@ def func_to_test4():
 
 
 result_var_type4 = {
-    'a': NodeVarType.NUMPY_NDARRAY,
-    'b': NodeVarType.NUMPY_NDARRAY,
-    'c': NodeVarType.TENSOR,
-    'd': NodeVarType.TENSOR
+    'a': {NodeVarType.NUMPY_NDARRAY},
+    'b': {NodeVarType.NUMPY_NDARRAY},
+    'c': {NodeVarType.TENSOR},
+    'd': {NodeVarType.TENSOR}
 }
 
-test_funcs = [func_to_test1, func_to_test2, func_to_test3, func_to_test4]
+
+def func_to_test5():
+    def inner_int_func():
+        return 1
+
+    def inner_bool_float_func(x):
+        a = 1.0
+        if x > 0:
+            return a
+        return False
+
+    def inner_unknown_func(x):
+        return x
+
+    a = inner_int_func()
+    b = inner_bool_float_func(3)
+    c = inner_unknown_func(None)
+    d = paddle.fluid.data('x', [1, 2])
+
+
+result_var_type5 = {
+    'a': {NodeVarType.INT},
+    'b': {NodeVarType.FLOAT, NodeVarType.BOOLEAN},
```

Comment:
Why the var type of `b` is not `{NodeVarType.FLOAT}` because `b` is 1.0

Reply:
Static analysis didn't run the python program. So the b's type is possible FLOAT or BOOLEAN before running.

Reply:
Or in another word, it is hard work to get the type by analyzing the control flow here :-)